### ISSUE TITLE
Flowing liquid z-fights with liquid source of different type below

### ISSUE THREAD

**Issue by Wuzzy2** [minetest/minetest#9323]
##### Minetest version
```
5.2.0-dev
```

##### Summary
If a flowing liquid of type A exists directly above a liquid source of type B, and you are inside the liquid source and look upwards to the flowing liquid, you will see z-fighting at the border between the 2 liquids.

Screenshot:
https://satoshiupload.com/images/pSjMidYgue.png

##### Steps to reproduce

This works for MTG.

1. First, create a lot of water source nodes (or go into the ocean)
2. Place a river water source on top so that it flows over the water sources
3. Go underwater and look upwards to the flowing river water nodes

**Comment by v-rob** [comment]
You and your z-fighting :-)

Is there any way to fix z-fighting either in Minetest's code or Irrlicht's so that it never happens and just one face is chosen or they are averaged somehow?  That would be much better than moving vertices since that is annoying (I can't use plantlike very well for precise things since it is offset) and won't work with large nodeboxes or meshes.

**Comment by Wuzzy2** [comment]
I have dug a little in the Irrlicht documentation and forums, and the recommended way to deal with this stuff is apparently using `irr::video::SMaterial::PolygonOffsetDirection` and `irr::video::SMaterial::PolygonOffsetFactor`. The forums told me this is the depth bias.

http://irrlicht.sourceforge.net/docu/classirr_1_1video_1_1_s_material.html#ab3da2285d0e74fc99eafdd8944ea4eb8
http://irrlicht.sourceforge.net/docu/classirr_1_1video_1_1_s_material.html#afa24a258ca0ca5eff3b66a3f8aa04462

**Comment by paramat** [comment]
Yes we should certainly eventually use depth bias and remove the position offset of plantlike.
